src/core/control/tools/EllipseHandler.cpp

```diff
@@ -63,8 +63,16 @@ auto EllipseHandler::createShape(bool isAltDown, bool isShiftDown, bool isContro
     /*
      * Set resolution depending on the radius (heuristic)
      */
-    auto nbPtsPerQuadrant =
-            static_cast<unsigned int>(std::ceil(5 + 0.2 * std::max(std::abs(radiusX), std::abs(radiusY))));
+    unsigned int nbPtsPerQuadrant;
+
+    const double excentricity = std::abs(radiusY / radiusX);
+
+
+    // Change the formula used to compute the points to construct the ellipse on the basis of the ellipse's excentricity 
+    if (excentricity < 0.4)
```

Comment:
If you see `0 < excentricity < 0.4` as  flattened (horizontally),  then `2.5 < excentricity` is also flattened (vertically). But as I wrote above, there is no need to separate cases.